sdk/core/core/src/az_http_policy.c

```diff
@@ -92,8 +96,35 @@ AZ_NODISCARD az_result az_http_pipeline_policy_logging(
     az_http_response * const response) {
   (void)data;
 
-  // Authentication logic
-  return az_http_pipeline_nextpolicy(p_policies, hrb, response);
+  if (data == NULL) {
+    return az_http_pipeline_nextpolicy(p_policies, hrb, response);
+  }
+
+  clock_t const threshold
+      = (data == NULL) ? 0 : ((az_log_options const *)data)->slow_response_threshold;
+  clock_t const start = clock();
+
+  az_result const result = az_http_pipeline_nextpolicy(p_policies, hrb, response);
+
+  clock_t const end = clock();
+
+  if (data != NULL) {
+    clock_t const duration = end - start;
+    if (duration >= threshold && az_log_should_write(AZ_LOG_SLOW_RESPONSE)) {
+      uint8_t log_msg_buf[200];
+      az_span_builder log_msg_bldr
+          = az_span_builder_create((az_mut_span)AZ_SPAN_FROM_ARRAY(log_msg_buf));
+
+      AZ_RETURN_IF_FAILED(az_span_builder_append_unsigned_number(
+          &log_msg_bldr, duration / (CLOCKS_PER_SEC / 1000)));
+
+      AZ_RETURN_IF_FAILED(az_span_builder_append(&log_msg_bldr, AZ_STR(" ms\0")));
+
+      az_log_write(AZ_LOG_SLOW_RESPONSE, (char const *)az_span_builder_result(&log_msg_bldr).begin);
```

Comment:
wondering..  can we have logging api to work with spans instead of char ?   
It might be less work to delegate converting span to 0-terminated str to logging, instead of taking care of this conversion from everywhere where we want to log something from one span builder.

Otherwise, why can't we just use a char array here directly instead of constructing the span?

Reply:
Done.